### PR TITLE
AutoGPTQ Mistral, Memory profiling support and empirical quality checks

### PR DESCRIPTION
**Pull request by Anindyadeep**
This PR introduces all the changes by PR #167 and integrates those in AutoGPTQ. AutoGPTQ README now has quality checks table for both Llama 2 and Mistral

### CODE REVIEW

File: bench_autogptq/bench.sh [hunk 1]
```diff
@@ -2,37 +2,40 @@
...
 print_usage() {
     echo "Usage: $0 [OPTIONS]"
     echo "OPTIONS:"
     echo "  -p, --prompt        Prompt for benchmarks (default: 'Write an essay about the transformer model architecture')"
     echo "  -r, --repetitions   Number of repetitions for benchmarks (default: 10)"
     echo "  -m, --max_tokens    Maximum number of tokens for benchmarks (default: 512)"
     echo "  -d, --device        Device for benchmarks (possible values: 'metal', 'cuda', and 'cpu', default: 'cuda')"
+    echo "  -n, --model_name    The name of the model to benchmark (possible values: 'llama' for using Llama2, 'mistral' for using Mistral 7B v0.1)"
     echo "  -lf, --log_file     Logging file name."
     echo "  -md, --models_dir   Models directory."
```
Comment (nsosio) [inline bench_autogptq/bench.sh:30]: do we need the model_dir?

File: bench_autogptq/bench.sh [hunk 2]
```diff
@@ -70,38 +73,34 @@ check_python() {
 
 
 setup() {
-
-    # Check if Logs folder exists else Make the logs folder
+    local MODEL_NAME="${1:-llama}"
     LOGS_FOLDER="$CURRENT_DIR/Logs"
```
Comment (nsosio) [inline bench_autogptq/bench.sh:77]: logs instead of Logs

File: bench_autogptq/bench.sh [hunk 3]
```diff
@@ -137,12 +136,8 @@ while [ "$#" -gt 0 ]; do
...
             shift 2
             ;;
-        -lf|--log_file)
-            LOG_FILENAME="$2"
-            shift 2
-            ;;
-        -md|--models_dir)
-            MODELS_DIR="$2"
+        -n|--model_name)
+            MODEL_NAME="$2"
```
Comment (nsosio) [inline bench_autogptq/bench.sh:140]: check supported models (e.g. only llama and mistral)

File: bench_autogptq/requirements.txt [hunk 4]
```diff
@@ -1,5 +1,6 @@
-torch==2.1.1
-accelerate
-transformers
-optimum
-auto_gptq
+numpy==1.26.4
+gekko==1.1.0
+pandas==2.2.1
+huggingface_hub
```
Comment (nsosio) [inline bench_autogptq/requirements.txt:4]: let's pin also this version

File: bench_autogptq/setup.sh [hunk 5]
```diff
@@ -27,28 +30,82 @@ check_python() {
...
+        exit 1
+    fi
+
+    "$PYTHON_CMD" -m pip install --upgrade pip > /dev/null
+    "$PYTHON_CMD" -m pip install -r "$SCRIPT_DIR/requirements.txt" --no-cache-dir > /dev/null
+
+    "$PYTHON_CMD" -m pip uninstall -y fsspec
+
+    # Install the required version of fsspec
+    "$PYTHON_CMD" -m pip install 'fsspec[http]>=2023.1.0,<=2024.2.0'
```
Comment (nsosio) [inline bench_autogptq/setup.sh:96]: why not moving this to the requirements.txt?

**Review by nsosio — COMMENTED** [review]
the result in the markdown file will be added later?

File: bench_autogptq/bench.sh (hunk 1 above)
Comment (Anindyadeep) [reply bench_autogptq/bench.sh:30]: Nope, need to delete that

File: bench_autogptq/bench.sh (hunk 3 above)
Comment (Anindyadeep) [reply bench_autogptq/bench.sh:140]: This thing is supported inside bench.py base class :)

https://github.com/premAI-io/benchmarks/blob/4515ac19ea2d7d075b1a98df14a22b433cb3e25e/common/base.py#L34-L36

File: bench_autogptq/requirements.txt (hunk 4 above)
Comment (Anindyadeep) [reply bench_autogptq/requirements.txt:4]: Done

File: bench_autogptq/bench.sh (hunk 2 above)
Comment (Anindyadeep) [reply bench_autogptq/bench.sh]: Done

Comment (Anindyadeep) [reply bench_autogptq/bench.sh]: Done

File: bench_autogptq/bench.sh (hunk 1 above)
Comment (Anindyadeep) [reply bench_autogptq/bench.sh]: Deleted this one

File: bench_autogptq/setup.sh (hunk 5 above)
Comment (Anindyadeep) [reply bench_autogptq/setup.sh:96]: Yeah, so even after I added this in requirements I am getting an error and the benchmarking process keeps getting broken. So I had to revert back to add this explicitly.

**Comment by nsosio** [comment]
@Anindyadeep can you solve conflicts?

**Comment by Anindyadeep** [comment]
> @Anindyadeep can you solve conflicts?

Yes, will ping you once that done

**Comment by Anindyadeep** [comment]
This is one is ready to merge 

cc: @nsosio :)

File: bench_autogptq/bench.sh [hunk 6]
```diff
@@ -2,37 +2,37 @@
 
 ########################################################################################################
 # Script: bench.sh
-# Description: This script runs benchmarks AutoGPTQ Llama benchmark.
+# Description: This script runs benchmarks AutoAWQ llama benchmark.
```
Comment (nsosio) [inline bench_autogptq/bench.sh:5]:
```suggestion
# Description: This script runs benchmarks AutoGPTQ benchmark.
```

**Pull request merged**